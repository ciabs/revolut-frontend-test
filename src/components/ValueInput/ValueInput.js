import React from 'react';
import PropTypes from 'prop-types';

import {Input} from './ValueInput.styles';

const ValueInput = ({name, value, onChange, disabled, isFromValue}) => (
  <Input
    name={name}
    type="number"
    min="0"
    value={value}
    onChange={onChange}
    disabled={disabled}
    isFromValue={isFromValue}
    isZero={value === 0}
  />
);

ValueInput.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired,
  disabled: PropTypes.bool.isRequired,
  isFromValue: PropTypes.bool.isRequired
};

export default ValueInput;
