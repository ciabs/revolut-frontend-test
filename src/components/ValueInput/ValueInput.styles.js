import styled from 'styled-components';
import {fromBackgroundColor, mainColor, toBackgroundColor, zeroValueColor} from '../../styles/constants';

export const Input = styled.input`
  text-align: right;
  border: none;
  outline: none;
  background-color: ${props => props.isFromValue ? fromBackgroundColor : toBackgroundColor};
  color: ${props => props.isZero ? zeroValueColor : mainColor};
  font-size: 30px;
  width: 100%;
`;
