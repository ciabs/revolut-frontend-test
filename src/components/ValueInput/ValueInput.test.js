import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ValueInput from './ValueInput';

describe('ValueInput', () => {
  const props = {
    name: 'fromValue',
    value: 0.15,
    onChange: jest.fn(),
    disabled: false,
    isFromValue: true
  };

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <ValueInput {...props} />,
      div
    );
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <ValueInput {...props} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
