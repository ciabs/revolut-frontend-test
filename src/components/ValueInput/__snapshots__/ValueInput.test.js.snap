// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValueInput has a valid snapshot 1`] = `
<input
  className="sc-bdVaJa ieueMs"
  disabled={false}
  min="0"
  name="fromValue"
  onChange={[Function]}
  placeholder="0"
  type="number"
  value={-0.15}
/>
`;
